### ISSUE TITLE
Popup menus shown by wxTaskBarIcon don't show icons any longer under Mac

### ISSUE THREAD

**Issue by vadz** [wxWidgets/wxWidgets#22188]
Reproducible with
```diff
diff --git a/samples/taskbar/tbtest.cpp b/samples/taskbar/tbtest.cpp
index 6a1f957343..bd903f3474 100644
--- a/samples/taskbar/tbtest.cpp
+++ b/samples/taskbar/tbtest.cpp
@@ -29,6 +29,8 @@

 #include "wx/taskbar.h"

+#include "wx/artprov.h"
+
 #include "tbtest.h"

 // ----------------------------------------------------------------------------
@@ -249,7 +251,8 @@ wxMenu *MyTaskBarIcon::CreatePopupMenu()
     wxMenu *menu = new wxMenu;
     menu->Append(PU_RESTORE, "&Restore main window");
     menu->AppendSeparator();
-    menu->Append(PU_NEW_ICON, "&Set New Icon");
+    menu->Append(PU_NEW_ICON, "&Set New Icon")
+        ->SetBitmap(wxArtProvider::GetBitmapBundle(wxART_REFRESH, wxART_MENU));
     menu->AppendSeparator();
     menu->AppendCheckItem(PU_CHECKMARK, "Test &check mark");
     menu->AppendSeparator();
```
in the taskbar sample.

See https://github.com/wxWidgets/wxWidgets/commit/337940f0091dcbc331affaebe0aba7c73d1a73cc

cc @dkulp @kosh543

**Comment by dkulp** [comment]
Quick patch:

```
diff --git a/src/osx/cocoa/taskbar.mm b/src/osx/cocoa/taskbar.mm
index 4685d516ce..79c478591d 100644
--- a/src/osx/cocoa/taskbar.mm
+++ b/src/osx/cocoa/taskbar.mm
@@ -269,15 +269,20 @@ WX_NSMenu wxTaskBarIconDockImpl::OSXDoGetDockHMenu()
 {
     m_menuDeleter.reset();

-    m_pMenu = GetPopupMenu();
+    wxMenu *menu = GetPopupMenu();
+    if (m_pMenu != menu)
+    {
+        m_pMenu = menu;
+        m_pMenu->SetupBitmaps();
+    }

     if (!m_pMenu)
     {
         m_pMenu = CreatePopupMenu();
-
         if (!m_pMenu)
             return nil;

+        m_pMenu->SetupBitmaps();
         m_menuDeleter.reset(m_pMenu);
     }

@@ -403,7 +408,7 @@ bool wxTaskBarIconCustomStatusItemImpl::RemoveIcon()
 bool wxTaskBarIconCustomStatusItemImpl::PopupMenu(wxMenu *menu)
 {
     wxASSERT(menu);
-
+    menu->SetupBitmaps();
     menu->SetInvokingWindow(m_eventWindow);
     menu->UpdateUI();

```

**Comment by vadz** [comment]
It looks like the diff above should work, but it doesn't work for me (macOS 12.0.1) and I don't see why -- I do see `wxMenuItemCocoaImpl::SetBitmap()` being called with a valid bitmap, but somehow it just doesn't appear on screen.

FWIW the following patch, which I came up with myself, and which tries to actually use the correct scale for the bitmap, doesn't work neither:
```diff
diff --git a/src/osx/cocoa/taskbar.mm b/src/osx/cocoa/taskbar.mm
index 4685d516ce..c546e28668 100644
--- a/src/osx/cocoa/taskbar.mm
+++ b/src/osx/cocoa/taskbar.mm
@@ -40,6 +40,8 @@
 public:
     wxTaskBarIconWindow(wxTaskBarIconImpl *impl);

+    double GetContentScaleFactor() const wxOVERRIDE;
+
     void OnMenuEvent(wxCommandEvent& event);
     void OnUpdateUIEvent(wxUpdateUIEvent& event);

@@ -282,7 +284,7 @@ wxBitmap IconFromBundle(const wxBitmapBundle& bndl)
     }

     m_pMenu->SetInvokingWindow(m_eventWindow);
-
+    m_pMenu->SetupBitmaps();
     m_pMenu->UpdateUI();

     return (WX_NSMenu)m_pMenu->GetHMenu();
@@ -405,6 +407,7 @@ - (wxTaskBarIconCustomStatusItemImpl*) implementation
     wxASSERT(menu);

     menu->SetInvokingWindow(m_eventWindow);
+    menu->SetupBitmaps();
     menu->UpdateUI();

     [m_statusItem popUpStatusItemMenu:(NSMenu*)menu->GetHMenu()];
@@ -427,6 +430,13 @@ - (wxTaskBarIconCustomStatusItemImpl*) implementation
 {
 }

+double wxTaskBarIconWindow::GetContentScaleFactor() const
+{
+    // We don't have any real window here, so use the main screen scale, which
+    // should be appropriate for dock/status icons.
+    return wxOSXGetMainScreenContentScaleFactor();
+}
+
 void wxTaskBarIconWindow::OnMenuEvent(wxCommandEvent& event)
 {
     m_impl->GetTaskBarIcon()->ProcessEvent(event);
```

**Comment by vadz** [comment]
Sorry, I don't know what to do here. I've even checked that the icon was valid under the debugger:
```
(lldb) po m_osxMenuItem.image
<NSImage 0x10553fac0 Size={16, 16} RepProvider=<NSImageArrayRepProvider: 0x104d25550, reps:(
    "<NSCGImageSnapshotRep:0x10454b180 cgImage=<CGImage 0x105e3f940> (DP)\n\t<<CGColorSpace 0x104705420> (kCGColorSpaceICCBased; kCGColorSpaceModelRGB; sRGB IEC61966-2.1)>\n\t\twidth = 32, height = 32, bpc = 8, bpp = 32, row bytes = 128 \n\t\tkCGImageAlphaPremultipliedFirst | 0 (default byte order)  | kCGImagePixelFormatPacked \n\t\tis mask? No, has masking color? No, has soft mask? No, has matte? No, should interpolate? Yes>"
)>>
```
and I just don't know why is it not shown.

Moreover, reverting 337940f0091dcbc331affaebe0aba7c73d1a73cc and applying
```diff
diff --git a/samples/taskbar/tbtest.cpp b/samples/taskbar/tbtest.cpp
index b91ebb6134..0a29523978 100644
--- a/samples/taskbar/tbtest.cpp
+++ b/samples/taskbar/tbtest.cpp
@@ -30,6 +30,8 @@
 #include "wx/artprov.h"
 #include "wx/taskbar.h"

+#include "wx/artprov.h"
+
 #include "tbtest.h"

 // ----------------------------------------------------------------------------
@@ -250,7 +252,8 @@ wxMenu *MyTaskBarIcon::CreatePopupMenu()
     wxMenu *menu = new wxMenu;
     menu->Append(PU_RESTORE, "&Restore main window");
     menu->AppendSeparator();
-    menu->Append(PU_NEW_ICON, "&Set New Icon");
+    menu->Append(PU_NEW_ICON, "&Set New Icon")
+        ->SetBitmap(wxArtProvider::GetBitmap(wxART_REFRESH, wxART_MENU));
     menu->AppendSeparator();
     menu->AppendCheckItem(PU_CHECKMARK, "Test &check mark");
     menu->AppendSeparator();
```
still doesn't show the icon in the sample for me, so it doesn't look like this bug is due to `wxBitmapBundle`-related changes at all.

@dkulp What was the version in which the sample actually worked for you? Or do you do something different from the sample which does work with your patch above (and then hopefully with the latest master, where I pushed my changes similar to what you did)?

In any case, I can't really do anything about this (at least not without some help/hints from @csomor), so this won't be fixed in 3.1.6, sorry.

**Comment by csomor** [comment]
@vadz @dkulp when testing this against master running under macOS 12.3.1 with an Apple Silicon Retina machine I get a proper refresh icon. Do you continue to see the problem and if yes, under which configs ? Thanks, STefa

**Comment by dkulp** [comment]
I think it was fixed as part of 616e7c88422bca5c5fa549e53836dd1b8e952a61 

In any case, they look OK now.

**Comment by csomor** [comment]
Thanks for the swift response.
